### PR TITLE
Upgrade some dependencies version

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Comment by chibenwa** [comment]
Here are candidates for upgrade

```
[INFO]   ch.qos.logback:logback-classic ............... 1.2.10 -> 1.2.11
[INFO]   com.fasterxml.jackson.core:jackson-annotations ...... 2.13.2 -> 2.13.3
[INFO]   com.fasterxml.jackson.core:jackson-core ............. 2.13.2 -> 2.13.3
[INFO]   com.fasterxml.jackson.core:jackson-databind ....... 2.13.2.2 -> 2.13.3
[INFO]   com.fasterxml.jackson.datatype:jackson-datatype-guava ...
[INFO]                                                         2.13.2 -> 2.13.3
[INFO]   com.fasterxml.jackson.datatype:jackson-datatype-jdk8 ...
[INFO]                                                         2.13.2 -> 2.13.3
[INFO]   com.fasterxml.jackson.datatype:jackson-datatype-jsr310 ...
[INFO]                                                         2.13.2 -> 2.13.3
[INFO]   com.github.ben-manes.caffeine:caffeine ................ 3.0.5 -> 3.1.0
[INFO]   com.github.dpaukov:combinatoricslib3 .................. 3.3.2 -> 3.3.3
[INFO]   com.google.guava:guava ........................ 31.0.1-jre -> 31.1-jre
[INFO]   com.google.inject:guice ............................... 5.0.1 -> 5.1.0
[INFO]   com.google.inject.extensions:guice-assistedinject ..... 5.0.1 -> 5.1.0
[INFO]   com.ibm.icu:icu4j ....................................... 70.1 -> 71.1
[INFO]   com.jayway.jsonpath:json-path ......................... 2.6.0 -> 2.7.0
[INFO]   com.unboundid:unboundid-ldapsdk ....................... 6.0.3 -> 6.0.5
[INFO]   commons-daemon:commons-daemon ......................... 1.2.4 -> 1.3.1
[INFO]   dnsjava:dnsjava ....................................... 3.4.3 -> 3.5.1
[INFO]   eu.timepit:refined_2.13 ............................. 0.9.28 -> 0.9.29
[INFO]   io.dropwizard.metrics:metrics-core .................... 4.2.7 -> 4.2.9
[INFO]   io.dropwizard.metrics:metrics-jmx ..................... 4.2.7 -> 4.2.9
[INFO]   io.dropwizard.metrics:metrics-jvm ..................... 4.2.7 -> 4.2.9
[INFO]   io.github.hakky54:sslcontext-kickstart-for-pem ........ 7.2.0 -> 7.4.3
[INFO]   io.jsonwebtoken:jjwt-api ............................ 0.11.2 -> 0.11.5
[INFO]   io.jsonwebtoken:jjwt-impl ........................... 0.11.2 -> 0.11.5
[INFO]   io.jsonwebtoken:jjwt-jackson ........................ 0.11.2 -> 0.11.5
[INFO]   io.rest-assured:rest-assured .......................... 4.4.0 -> 5.1.0
[INFO]   net.javacrumbs.json-unit:json-unit-assertj .......... 2.28.0 -> 2.35.0
[INFO]   nl.jqno.equalsverifier:equalsverifier .................. 3.8.2 -> 3.10
[INFO]   org.apache.activemq:activemq-broker ................. 5.16.3 -> 5.17.1
[INFO]   org.apache.activemq:artemis-jms-client .............. 2.20.0 -> 2.22.0
[INFO]   org.apache.logging.log4j:log4j ...................... 2.17.1 -> 2.17.2
[INFO]   org.apache.logging.log4j:log4j-slf4j-impl ........... 2.17.1 -> 2.17.2
[INFO]   org.apache.maven:maven-plugin-api ..................... 3.8.4 -> 3.8.5
[INFO]   org.apache.maven.plugin-tools:maven-plugin-annotations ...
[INFO]                                                           3.6.2 -> 3.6.4
[INFO]   org.apache.pdfbox:pdfbox ......................... 2.0.25 -> 2.0.26
[INFO]   org.apache.xbean:xbean-spring ........................... 4.18 -> 4.21
[INFO]   org.awaitility:awaitility ............................. 4.1.1 -> 4.2.0
[INFO]   org.jsoup:jsoup ..................................... 1.14.3 -> 1.15.1
[INFO]   org.junit.jupiter:junit-jupiter-engine ............. 5.7.0 -> 5.8.2
[INFO]   org.junit.jupiter:junit-jupiter-migrationsupport ... 5.7.0 -> 5.8.2
[INFO]   org.junit.jupiter:junit-jupiter-params ............. 5.7.0 -> 5.8.2
[INFO]   org.junit.platform:junit-platform-engine ........... 1.7.0 -> 1.8.2
[INFO]   org.junit.platform:junit-platform-launcher ......... 1.7.0 -> 1.8.2
[INFO]   org.junit.vintage:junit-vintage-engine ............. 5.7.0 -> 5.8.2
[INFO]   org.mock-server:mockserver-netty .................... 5.11.2 -> 5.13.2
[INFO]   org.mockito:mockito-core .............................. 3.7.7 -> 4.5.1
[INFO]   org.scala-lang:scala-library ........................ 2.13.7 -> 2.13.8
[INFO]   org.scalacheck:scalacheck_2.13 ...................... 1.15.4 -> 1.16.0
[INFO]   org.scalatest:scalatest_2.13 ................... 3.2.10 -> 3.2.12
[INFO]   org.slf4j:jcl-over-slf4j ...................... 1.7.32 -> 1.7.36
[INFO]   org.slf4j:log4j-over-slf4j .................... 1.7.32 -> 1.7.36
[INFO]   org.slf4j:slf4j-api ........................... 1.7.32 -> 1.7.36
[INFO]   org.testcontainers:junit-jupiter .................... 1.16.2 -> 1.17.2
[INFO]   org.testcontainers:pulsar ........................... 1.16.2 -> 1.17.2
[INFO]   org.testcontainers:testcontainers ................... 1.16.2 -> 1.17.2
```

**Comment by Arsnael** [comment]
I'm doing the upgrade of jackson to 2.13.3 already in ES upgrade btw, as it's used by es 8.2 java client api

File: pom.xml [hunk 1]
```diff
@@ -3351,7 +3351,7 @@
                 <plugin>
                     <groupId>org.codehaus.mojo</groupId>
                     <artifactId>versions-maven-plugin</artifactId>
-                    <version>2.7</version>
+                    <version>2.11.0</version>
```
Comment (Arsnael) [inline pom.xml:3356]: Commit says 2.10.0 but here it's 2.11.0... Maybe change the commit description to match the right version :)

File: backends-common/elasticsearch-v7/pom.xml [hunk 2]
```diff
@@ -87,12 +87,16 @@
         <dependency>
             <groupId>org.elasticsearch.client</groupId>
             <artifactId>elasticsearch-rest-high-level-client</artifactId>
-            <version>7.10.2</version>
+            <version>7.17.3</version>
```
Comment (Arsnael) [inline backends-common/elasticsearch-v7/pom.xml:90]: Not part of your commit description, and unnecessary as we try to upgrade ES to 8 and that dependency disappears, replaced by an other client

**Review by Arsnael — COMMENTED** [review]
Good job overall. Don't forget to squash the fixup you have hiding in the middle too :)

**Comment by Arsnael** [comment]
```
02:28:56,646 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project apache-james-backends-es-v7: Compilation failure: Compilation failure: 

02:28:56,646 [ERROR] /home/jenkins/jenkins-agent/workspace/james_ApacheJames_PR-1017/backends-common/elasticsearch-v7/src/main/java/org/apache/james/backends/es/v7/ElasticSearchIndexer.java:[34,41] cannot find symbol

02:28:56,646 [ERROR]   symbol:   class XContentType
```
Yeah definitely do not bother upgrading the es client once again, as we are getting rid of it with the es upgrade anyways

File: backends-common/elasticsearch-v7/pom.xml (hunk 2 above)
Comment (vttranlina) [reply backends-common/elasticsearch-v7/pom.xml:90]: yes, my miss, I see your pr.

File: server/container/spring/pom.xml [hunk 3]
```diff
@@ -164,6 +164,11 @@
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-slf4j-impl</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (chibenwa) [inline server/container/spring/pom.xml:171]: Why is this needed?

File: server/container/core/pom.xml [hunk 4]
```diff
@@ -102,6 +102,11 @@
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (chibenwa) [inline server/container/core/pom.xml:109]: Why is this needed?

Comment (vttranlina) [reply server/container/core/pom.xml:109]: image

https://github.com/apache/james-project/pull/1017/commits/1eba0e43271725543ef5720cbe7945116686c2a8

**Review by quantranhong1999 — COMMENTED** [review]
Read it. Nothing to add.

**Comment by vttranlina** [comment]
```
org.opentest4j.AssertionFailedError: 

Expecting value to be false but was true
	at org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPoolTest.notUsedChannelShouldBeClosedWhenPoolIsClosed(ReactorRabbitMQChannelPoolTest.java:117)
``` 
I tried to re-test on my local, it works. 
I think it is a CI problem. 
I will rebase the latest master

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Related?

```
<p><span style="font-weight:bold">org.apache.james.mailbox.MailboxManagerTest$HookTests$PreDeletion.expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted</span>
          (from org.apache.james.mailbox.cassandra.CassandraMailboxManagerTest)
        </p><div style="text-align:right;">Failing for the past 1 build
          (Since <a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/" class="model-link"><span style="width: 16px; height: 16px; " class="build-status-icon__wrapper icon-red icon-sm"><span class="build-status-icon__outer"><svg viewBox="0 0 24 24" focusable="false" class="svg-icon "><use href="https://ci-builds.apache.org/images/build-status/build-status-sprite.svg#build-status-static"></use></svg></span><svg viewBox="0 0 24 24" focusable="false" class="svg-icon icon-red icon-sm"><use href="https://ci-builds.apache.org/static/b3a0cd98/images/build-status/build-status-sprite.svg#last-failed"></use></svg></span>#7</a> )
        </div><div style="text-align:right;"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/testReport/junit/org.apache.james.mailbox/MailboxManagerTest$HookTests$PreDeletion/expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted/history">Took 0.73 sec.</a></div><div id="description"><div></div><div class="jenkins-buttons-row jenkins-buttons-row--invert"></div></div>
<h3>Error Message</h3><pre>org.mockito.Mockito.verifyZeroInteractions([Ljava/lang/Object;)V</pre><h3>Stacktrace</h3><pre>java.lang.NoSuchMethodError: org.mockito.Mockito.verifyZeroInteractions([Ljava/lang/Object;)V
	at org.apache.james.mailbox.MailboxManagerTest$HookTests$PreDeletion.expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted(MailboxManagerTest.java:2719)
</pre>org.apache.james.mailbox.MailboxManagerTest$HookTests$PreDeletion.expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted (from org.apache.james.mailbox.cassandra.CassandraMailboxManagerTest)
Failing for the past 1 build (Since [#7](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/) )
[Took 0.73 sec.](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/testReport/junit/org.apache.james.mailbox/MailboxManagerTest$HookTests$PreDeletion/expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted/history)
Error Message

org.mockito.Mockito.verifyZeroInteractions([Ljava/lang/Object;)V

Stacktrace

java.lang.NoSuchMethodError: org.mockito.Mockito.verifyZeroInteractions([Ljava/lang/Object;)V
	at org.apache.james.mailbox.MailboxManagerTest$HookTests$PreDeletion.expungeShouldNotCallPredeletionHooksWhenNoMessagesMarkedAsDeleted(MailboxManagerTest.java:2719)
```

```
<p><span style="font-weight:bold">org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken</span>
          (from org.apache.james.imapserver.netty.IMAPServerTest)
        </p><div style="text-align:right;">Failing for the past 1 build
          (Since <a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/" class="model-link"><span style="width: 16px; height: 16px; " class="build-status-icon__wrapper icon-red icon-sm"><span class="build-status-icon__outer"><svg viewBox="0 0 24 24" focusable="false" class="svg-icon "><use href="https://ci-builds.apache.org/images/build-status/build-status-sprite.svg#build-status-static"></use></svg></span><svg viewBox="0 0 24 24" focusable="false" class="svg-icon icon-red icon-sm"><use href="https://ci-builds.apache.org/static/b3a0cd98/images/build-status/build-status-sprite.svg#last-failed"></use></svg></span>#7</a> )
        </div><div style="text-align:right;"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/testReport/junit/org.apache.james.imapserver.netty/IMAPServerTest$Oidc/oauthShouldSuccessWhenValidToken/history">Took 0.51 sec.</a></div><div id="description"><div></div><div class="jenkins-buttons-row jenkins-buttons-row--invert"></div></div>
<h3>Error Message</h3><pre>org/apache/james/protocols/api/OIDCSASLHelper</pre><h3>Stacktrace</h3><pre>java.lang.NoClassDefFoundError: org/apache/james/protocols/api/OIDCSASLHelper
	at org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken(IMAPServerTest.java:1049)
Caused by: java.lang.ClassNotFoundException: org.apache.james.protocols.api.OIDCSASLHelper
	at org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken(IMAPServerTest.java:1049</pre>org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken (from org.apache.james.imapserver.netty.IMAPServerTest)
Failing for the past 1 build (Since [#7](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/) )
[Took 0.51 sec.](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1017/7/testReport/junit/org.apache.james.imapserver.netty/IMAPServerTest$Oidc/oauthShouldSuccessWhenValidToken/history)
Error Message

org/apache/james/protocols/api/OIDCSASLHelper

Stacktrace

java.lang.NoClassDefFoundError: org/apache/james/protocols/api/OIDCSASLHelper
	at org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken(IMAPServerTest.java:1049)
Caused by: java.lang.ClassNotFoundException: org.apache.james.protocols.api.OIDCSASLHelper
	at org.apache.james.imapserver.netty.IMAPServerTest$Oidc.oauthShouldSuccessWhenValidToken(IMAPServerTest.java:1049
```

Could you have a look?

**Comment by vttranlina** [comment]
> java.lang.NoSuchMethodError: org.mockito.Mockito.verifyZeroInteractions([Ljava/lang/Object;)V

Yes, it is related, `verifyZeroInteractions` is the new feature of the new mockito. 
But I checked on my local, and it works.

**Comment by vttranlina** [comment]
I tried to run `mvn test` on local for all module ci failed, 
My maven builds success.

**Comment by chibenwa** [comment]
Rebase maybe?

**Comment by vttranlina** [comment]
> Rebase maybe?

it already

**Comment by Arsnael** [comment]
You ran it locally with maven or the IDE? you have a difference sometimes :) try with maven

**Comment by vttranlina** [comment]
It green :D
